### PR TITLE
C Code Cleanup

### PR DESCRIPTION
**Pull request by phudtran**
Clean up of unuseded code from Mizar.
Also added .clang_format

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+#SpaceBeforeRangeBasedForLoopColon: true # Unknown to clang-format-5.0
+SpaceInEmptyParentheses: false
+SpacesBeforeTrailingComments: 1
+SpacesInAngles: false
+SpacesInContainerLiterals: false
+SpacesInCStyleCastParentheses: false
+SpacesInParentheses: false
+SpacesInSquareBrackets: false
+Standard: Cpp03
+TabWidth: 8
```
Comment (liangbin-pub) [inline .clang-format]: not familiar with the effect of this file, is it used to check at compile time or as a formatter for editor? Should we use Tab? should it be 8 spaces?

File: src/cli/test/test_cli.c [hunk 2]
```diff
@@ -751,570 +299,75 @@ static void test_trn_cli_load_transit_subcmd(void **state)
...
-	rc = trn_cli_load_transit_subcmd(NULL, argc, argv3);
-	assert_int_equal(rc, -EINVAL);
-
-	/* Test call load_transit_xdp return NULL*/
-	TEST_CASE("load_transit_xdp subcommand fails if rpc returns NULl");
-	expect_function_call(__wrap_load_transit_xdp_1);
-	will_return(__wrap_load_transit_xdp_1, NULL);
-	rc = trn_cli_load_transit_subcmd(NULL, argc, argv1);
-	assert_int_equal(rc, -EINVAL);
-}
```
Comment (liangbin-pub) [inline src/cli/test/test_cli.c:789]: Do we need to remove these transit xdp test cases?

File: src/cli/test/test_cli.c [hunk 3]
```diff
@@ -751,570 +299,75 @@ static void test_trn_cli_load_transit_subcmd(void **state)
...
-	rc = trn_cli_unload_transit_subcmd(NULL, argc, argv1);
-	assert_int_equal(rc, -EINVAL);
-
-	/* Test call unload_transit_xdp_! return NULL*/
-	TEST_CASE("unload_transit_xdp subcommand fails if rpc returns NULl");
-	expect_function_call(__wrap_unload_transit_xdp_1);
-	will_return(__wrap_unload_transit_xdp_1, NULL);
-	rc = trn_cli_unload_transit_subcmd(NULL, argc, argv1);
-	assert_int_equal(rc, -EINVAL);
-}
```
Comment (liangbin-pub) [inline src/cli/test/test_cli.c:824]: Same comment as above, are these transit xdp tests not needed?

File: src/cli/trn_cli_xdp.c [hunk 4]
```diff
@@ -210,92 +210,4 @@ int trn_cli_unload_transit_subcmd(CLIENT *clnt, int argc, char *argv[])
...
-			rpc);
-		return -EINVAL;
-	}
-
-	print_msg(
-		"unload_transit_agent_xdp_1 successfully unloaded transit xdp on interface %s.\n",
-		intf.interface);
-	return 0;
-}
+}
```
Comment (liangbin-pub) [inline src/cli/trn_cli_xdp.c]: newline

File: src/xdp/trn_transit_xdp.c [hunk 5]
```diff
@@ -559,14 +262,6 @@ static __inline int trn_process_inner_arp(struct transit_packet *pkt)
...
 	ep = bpf_map_lookup_elem(&endpoints_map, &epkey);
 
-	/* Don't respond to arp if endpoint is not found, or it is local to host */
-	if (!ep || ep->hosted_iface != -1 ||
-	    pkt->inner_arp->ar_op != bpf_htons(ARPOP_REQUEST)) {
-		bpf_debug("[Transit:%d:0x%x] Bypass ARP handling\n", __LINE__,
-			  bpf_ntohl(pkt->itf_ipv4));
-		return trn_switch_handle_pkt(pkt, *sip, *tip, *sip);
-	}
-
```
Comment (liangbin-pub) [inline src/xdp/trn_transit_xdp.c:569]: ARP responder logic should be kept. I'm working on this file as well so I can bring it back later

File: src/xdp/trn_transit_scaled_endpoint.c [hunk 6]
```diff
@@ -1,664 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (liangbin-pub) [inline src/xdp/trn_transit_scaled_endpoint.c:1]: Leave this file in, we need it

File: src/cli/test/test_cli.c (hunk 2 above)
Comment (phudtran) [reply src/cli/test/test_cli.c:789]: These tests cases are still in.
Just an issue with git diff I think.
It thinks I removed these code blocks and re-added them

File: src/cli/test/test_cli.c (hunk 3 above)
Comment (phudtran) [reply src/cli/test/test_cli.c:824]: Tests are still in
Just an issue with git diff I think that thinks I removed this code and re-added it.

File: .clang-format (hunk 1 above)
Comment (phudtran) [reply .clang-format]: Previously in Mizar we used this file  format  the C code.
(Currently I have autoformat on save setup for my vscode)

It looks like the line with tabwitdth does not actually do anything. I'll remove it.

All of the c code is currently formatted by this file way with tabs.
Do you think it's better if we change it to spaces?

**Review by liangbin-pub — APPROVED** [review]

File: src/xdp/trn_transit_xdp.c (hunk 5 above)
Comment (phudtran) [reply src/xdp/trn_transit_xdp.c:569]: Ok sounds good!

File: src/xdp/trn_transit_scaled_endpoint.c (hunk 6 above)
Comment (phudtran) [reply src/xdp/trn_transit_scaled_endpoint.c:1]: I think we can probably add it back later as this scaled endpoint implementation doesn't quite match with the Zeta's current scaled endpoint design

File: src/cli/trn_cli_xdp.c (hunk 4 above)
Comment (phudtran) [reply src/cli/trn_cli_xdp.c]: Good catch!

File: .clang-format (hunk 1 above)
Comment (liangbin-pub) [reply .clang-format]: It's better to use space (except in Makefile) always, but for existing files, let's don't make white space changes.

**Review by liangbin-pub — COMMENTED** [review]
LGTM

**Pull request merged**